### PR TITLE
fix: flaky test `BTC Account - Overview has balance`

### LINKED ISSUE #28182: Fix "BTC Account - Overview has balance" flaky test
### What is this about?

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/108574/workflows/4249e906-8122-43d4-bf25-81f75ac78497/jobs/4060900/tests#failed-test-0

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by seaona**
## **Description**
Anti-pattern where we assert the balance BTC balance is 1BTC using getText. We should rather find the element by the expected balance to avoid race conditions

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28181?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/28182

## **Manual testing steps**

1. Check ci

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/flask/btc/btc-account-overview.spec.ts [hunk 1]
```diff
@@ -46,17 +45,19 @@ describe('BTC Account - Overview', function (this: Suite) {
     await withBtcAccountSnap(
       { title: this.test?.fullTitle() },
       async (driver) => {
-        // Wait for the balance to load up
-        await driver.delay(2000);
```
Comment (seaona) [inline test/e2e/flask/btc/btc-account-overview.spec.ts:50]: no need to wait if we look for the element by the value we expect

File: test/e2e/flask/btc/btc-account-overview.spec.ts [hunk 2]
```diff
@@ -46,17 +45,19 @@ describe('BTC Account - Overview', function (this: Suite) {
...
-
-        const balanceElement = await driver.findElement(
-          '.coin-overview__balance',
-        );
-        const balanceText = await balanceElement.getText();
```
Comment (seaona) [inline test/e2e/flask/btc/btc-account-overview.spec.ts:55]: anti-pattern

**Review by DDDDDanica — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]
LGTM

File: test/e2e/flask/btc/btc-account-overview.spec.ts [hunk 3]
```diff
@@ -46,17 +45,19 @@ describe('BTC Account - Overview', function (this: Suite) {
...
+        await driver.waitForSelector({
+          testId: 'account-value-and-suffix',
```
Comment (hjetpoluru) [inline test/e2e/flask/btc/btc-account-overview.spec.ts:49]: Nice!

Comment (seaona) [reply test/e2e/flask/btc/btc-account-overview.spec.ts:49]: cc @HowardBraham

Comment (HowardBraham) [reply test/e2e/flask/btc/btc-account-overview.spec.ts:49]: Haha yay!

**Pull request merged**